hudi-common/src/main/java/org/apache/hudi/common/table/view/RocksDbBasedFileSystemView.java

```diff
@@ -269,41 +269,51 @@ protected Option<Pair<String, CompactionOperation>> getPendingLogCompactionOpera
 
   @Override
   protected boolean isPartitionAvailableInStore(String partitionPath) {
-    String lookupKey = schemaHelper.getKeyForPartitionLookup(partitionPath);
-    Serializable obj = rocksDB.get(schemaHelper.getColFamilyForStoredPartitions(), lookupKey);
-    return obj != null;
+    try {
+      readLock.lock();
```

Comment:
I don't think we should add any lock to protected scope methods.

Reply:
Why not? The point here is to guarantee thread-safety to the store.

Reply:
ditto

Reply:
Replied above